lib/pure/strutils.nim

```diff
@@ -1493,7 +1493,7 @@ func delete*(s: var string, slice: Slice[int]) =
     a.delete(1..<1) # empty slice
     assert a == "ad"
   when compileOption("boundChecks"):
-    if not (slice.a < s.len and slice.a >= 0 and slice.b < s.len):
+    if (slice.a > s.len) or (slice.a < 0) or (slice.b >= s.len):
```

Comment:
This one has a slight semantic change. An empty slice (`0 ..< 0` or `0 .. -1` is it is internally is now valid).

Reply:
Empty slice makes sense to me, we got a test for this?

Reply:
yes I added a test in `tstrutils` for it.

Reply:
The only change in logic here (because the rest is De Morgan's law) is that instead of checking `slice.a >= s.len` (`not slice.a < s.len` in the pre-PR code) now `slice.a > s.len` is checked.
So this PR makes `0..<0` or `0..<-1` valid for a delete operation on an empty sequence. On non-empty sequences they were already valid. As a side-effect this also makes `5..<4` valid on a sequence with length 4. `5..<5` on a sequence with length 4 will still be invalid though.
Just thought I'd put this clarification here